common/server_protocolhandler.cpp

```diff
@@ -620,23 +620,9 @@ Response::ResponseCode Server_ProtocolHandler::cmdJoinRoom(const Command_JoinRoo
     if (!r)
         return Response::RespNameNotFound;
 
-    QString roomPermission = r->getRoomPermission().toLower();
-    if (roomPermission != "none"){
-        if (roomPermission == "registered") {
-            if (!(userInfo->user_level() & ServerInfo_User::IsRegistered))
-                return Response::RespUserLevelTooLow;
-        }
-
-        if (roomPermission == "moderator"){
-            if (!(userInfo->user_level() & ServerInfo_User::IsModerator))
-                return Response::RespUserLevelTooLow;
-        }
-
-        if (roomPermission == "administrator"){
-            if (!(userInfo->user_level() & ServerInfo_User::IsAdmin))
-                return Response::RespUserLevelTooLow;
-        }
-    }
+    if (!(userInfo->user_level() & ServerInfo_User::IsModerator || userInfo->user_level() & ServerInfo_User::IsModerator))
```

Comment:
That is what its doing.  There is no double check.  See how the IsModertor || isAdmin is contained in () with an outer ! ().  Its sayi is not moderator or admin.